test/test_images/listener/main.go

```diff
@@ -32,18 +43,37 @@ func main() {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	// Launch a go routine to avoid blocking
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Fatalf("Unable to read environment config: %v", err)
+	}
+
+	numExpectedEvents, err := strconv.Atoi(env.ExpectedEventCount)
+	if err != nil {
+		log.Fatalf("Invalid value for EVENT_COUNT (%v): %v", env.ExpectedEventCount, err)
+	}
+
+	events := make(chan cloudevents.Event, numExpectedEvents)
+	// receive events, putting them into the channel only if they meet the type we are expecting
 	go func() {
-		timeout := 10 * time.Second
-		<-time.After(timeout)
-		logging.FromContext(ctx).Infow("cancelling context: timeout reached", "timeout", timeout)
-		cancel()
+		if err = client.StartReceiver(ctx, func(event cloudevents.Event) {
+			logging.FromContext(ctx).Infof("Received event: %s", event.String())
+			if event.Type() == env.ExpectedEventType {
+				events <- event
+			}
+		}); err != nil {
+			logging.FromContext(ctx).Fatalf("receiving events: %v", err)
```

Comment:
Not sure if you want to replace the `logging.FromContext(ctx)` calls here and below with a single `log := logging.FromContext(ctx)` above and then just say `log.Fatalf(...)` through here.

Reply:
oh good point, I didn't realize that was an option 👍